web-ng/src/app/services/data-store/data-store.service.ts

```diff
@@ -136,6 +136,19 @@ export class DataStoreService {
       );
   }
 
+  loadObservation$(project: Project, observationId: string) {
+    return this.db
+      .collection(`projects/${project.id}/observations`)
+      .doc(observationId)
+      .get()
+      .pipe(
+        map(doc => {
+          const form = project.getForm(doc.data()!.formId);
```

Comment:
The observation data returned from Firestore will have the ids necessary to rebuild the model graph; perhaps we can move this step into the converter, and use the layerId and formId inside the observation data to get the layer from the project, and the form from the layer, instead of searching for it as we in `getForm()` below? Firebase UUIDs are globally unique (statistically speaking), so this implementation would work w/Firebase as a backend, but might break if observationId weren't globally unique. It would also be a bit easier to follow logically, since searching through layers in `getForm` implies we don't which layer and observation belongs to, which isn't the case.

Reply:
done!